packages/venia-concept/src/components/FilterModal/FilterSearch/filterSearch.js

```diff
@@ -0,0 +1,88 @@
+import React, { Fragment } from 'react';
+import TextInput from 'src/components/TextInput';
+import Trigger from 'src/components/Trigger';
+import { Form } from 'informed';
+import Icon from 'src/components/Icon';
+import ClearIcon from 'react-feather/dist/icons/x';
+import SearchIcon from 'react-feather/dist/icons/search';
+import classify from 'src/classify';
+import defaultClasses from './filterSearch.css';
+
+const clearIcon = <Icon src={ClearIcon} size={18} />;
+const searchIcon = <Icon src={SearchIcon} size={18} />;
+
+const withFilterSearch = WrappedComponent => {
+    class withFilterSearch extends React.Component {
+        state = {
+            filterQuery: ''
+        };
+
+        handleFilterSearch = event => {
+            const { value } = event.currentTarget || event.srcElement;
+            this.setState({ filterQuery: value });
+        };
+
+        handleResetSearch = () => this.setState({ filterQuery: '' });
+
+        getFilteredItems = (items, filterQuery) =>
+            items.filter(
+                item =>
+                    item.label
+                        .toUpperCase()
+                        .indexOf(filterQuery.toUpperCase()) >= 0
+            );
+
+        render() {
+            const {
+                handleFilterSearch,
+                getFilteredItems,
+                handleResetSearch
+            } = this;
+            const { filterQuery } = this.state;
+            const { items, classes, options, name, ...rest } = this.props;
+
+            const isSearchable = options && options.searchable;
+
+            const filteredItems = isSearchable
+                ? getFilteredItems(items, filterQuery)
+                : items;
+
+            const resetButton = filterQuery && (
+                <Trigger action={handleResetSearch}>{clearIcon}</Trigger>
+            );
+
+            return (
+                <Fragment>
+                    {isSearchable && (
+                        <Form className={classes.filterSearch}>
+                            <TextInput
+                                placeholder={`Search for a specific ${name}`}
+                                fieldState={{ value: filterQuery }}
+                                onChange={handleFilterSearch}
```

Comment:
Consider using `onValueChange` here, so we don't have to interact with an event object or the DOM.

Reply:
Fixed in latest commit.